File: mnheme_ports/java/mnheme/Feeling.java
// mnheme/java/mnheme/Feeling.java
package mnheme;

public enum Feeling {
    GIOIA("gioia"), TRISTEZZA("tristezza"), RABBIA("rabbia"), PAURA("paura"),
    NOSTALGIA("nostalgia"), AMORE("amore"), MALINCONIA("malinconia"),
    SERENITA("serenità"), SORPRESA("sorpresa"), ANSIA("ansia"),
    GRATITUDINE("gratitudine"), VERGOGNA("vergogna"), ORGOGLIO("orgoglio"),
    NOIA("noia"), CURIOSITA("curiosità");

    private final String value;

    Feeling(String value) { this.value = value; }

    public String getValue() { return value; }

    public static Feeling fromString(String s) {
        for (Feeling f : values()) {
            if (f.value.equalsIgnoreCase(s)) return f;
        }
        throw new IllegalArgumentException("Unknown feeling: " + s);
    }

    public static boolean isValid(String s) {
        for (Feeling f : values())
            if (f.value.equalsIgnoreCase(s)) return true;
        return false;
    }
}

File: mnheme_ports/kotlin/Feeling.kt
// mnheme/kotlin/mnheme/Feeling.kt
package mnheme

enum class Feeling(val value: String) {
    GIOIA       ("gioia"),
    TRISTEZZA   ("tristezza"),
    RABBIA      ("rabbia"),
    PAURA       ("paura"),
    NOSTALGIA   ("nostalgia"),
    AMORE       ("amore"),
    MALINCONIA  ("malinconia"),
    SERENITA    ("serenità"),
    SORPRESA    ("sorpresa"),
    ANSIA       ("ansia"),
    GRATITUDINE ("gratitudine"),
    VERGOGNA    ("vergogna"),
    ORGOGLIO    ("orgoglio"),
    NOIA        ("noia"),
    CURIOSITA   ("curiosità");

    companion object {
        private val byValue = entries.associateBy { it.value }
        fun fromString(s: String): Feeling =
            byValue[s.lowercase()] ?: throw IllegalArgumentException("Unknown feeling: $s")
        fun isValid(s: String): Boolean = s.lowercase() in byValue
        val validValues: Set<String> get() = byValue.keys
    }
}
